runtime/integration-tests/src/sibling.rs

```diff
@@ -0,0 +1,557 @@
+	AccountId32,
+};
+use xcm::v3::prelude::*;
+use xcm_emulator::{
+	cumulus_pallet_parachain_system::{self, RelayNumberStrictlyIncreases},
+	Weight,
+};
+use xcm_executor::{
+	traits::{JustTry, ShouldExecute, WeightTrader},
+	Assets, XcmExecutor,
+};
+
+use spacewalk_primitives::CurrencyId;
+use xcm::latest::Weight as XCMWeight;
+use xcm_builder::{
+	AccountId32Aliases, AllowUnpaidExecutionFrom, ConvertedConcreteId, EnsureXcmOrigin,
+	FixedWeightBounds, FungiblesAdapter, NoChecking, ParentIsPreset, RelayChainAsNative,
+	SiblingParachainAsNative, SiblingParachainConvertsVia, SignedAccountId32AsNative,
+	SignedToAccountId32, SovereignSignedViaLocation,
+};
+
+pub type AccountId = AccountId32;
+
+parameter_types! {
+	pub const RelayLocation: MultiLocation = MultiLocation::parent();
+	pub const RelayNetwork: NetworkId = NetworkId::Polkadot;
+	pub RelayChainOrigin: RuntimeOrigin = cumulus_pallet_xcm::Origin::Relay.into();
+	pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+	pub UniversalLocation: InteriorMultiLocation =
+		X2(GlobalConsensus(RelayNetwork::get()), Parachain(ParachainInfo::parachain_id().into()));
+
+}
+
+/// Type for specifying how a `MultiLocation` can be converted into an `AccountId`. This is used
+/// when determining ownership of accounts for asset transacting and when attempting to use XCM
+/// `Transact` in order to determine the dispatch Origin.
+pub type LocationToAccountId = (
+	// The parent (Relay-chain) origin converts to the parent `AccountId`.
+	ParentIsPreset<AccountId>,
+	// Sibling parachain origins convert to AccountId via the `ParaId::into`.
+	SiblingParachainConvertsVia<Sibling, AccountId>,
+	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
+	AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+/// CurrencyIdConvert
+/// This type implements conversions from our `CurrencyId` type into `MultiLocation` and vice-versa.
+/// A currency locally is identified with a `CurrencyId` variant but in the network it is identified
+/// in the form of a `MultiLocation`, in this case a pCfg (Para-Id, Currency-Id).
+pub struct CurrencyIdConvert;
+
+// Only supports native currency for now
+impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
+		match id {
+			CurrencyId::Native => Some(MultiLocation::new(
+				1,
+				X2(Parachain(ParachainInfo::parachain_id().into()), PalletInstance(10)),
+			)),
+			_ => None,
+		}
+	}
+}
+
+impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(location: MultiLocation) -> Option<CurrencyId> {
+		match location {
+			// Just for testing purposes, parachain id is not verified so this can be used by all runtimes
+			MultiLocation { parents: 1, interior: X2(Parachain(_id), PalletInstance(10)) } =>
+				Some(CurrencyId::Native),
+			MultiLocation { parents: 0, interior: X1(PalletInstance(10)) } =>
+				Some(CurrencyId::Native),
+			_ => None,
```

Comment:
This means that any native tokens that we receive from another chain are mixed up with the native tokens we already have on the sibling chain, no?  I would prefer we make this conversion very explicit and use a new `CurrencyId` that is custom to the sibling chain.

I'd create a `CurrencyId` similar to this
```
pub enum CurrencyId {
 Relay,
 Pendulum,
 Amplitude
}
```

and then in the conversion functions we do something like:
```
impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
	fn convert(location: MultiLocation) -> Option<CurrencyId> {
              match location {
			MultiLocation { parents: 1, interior: X2(Parachain(PENDULUM_ID), PalletInstance(10)) } =>
				Some(CurrencyId::Pendulum),
                         MultiLocation { parents: 1, interior: X2(Parachain(AMPLITUDE_ID), PalletInstance(10)) } =>
				Some(CurrencyId::Amplitude),
			MultiLocation { parents: 0, interior: X1(PalletInstance(10)) } =>
				Some(CurrencyId::Native),
			_ => None,
```

and

```
impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
	fn convert(id: CurrencyId) -> Option<MultiLocation> {
		match id {
			CurrencyId::Native => Some(MultiLocation::new(
				1,
				X2(Parachain(ParachainInfo::parachain_id().into()), PalletInstance(10)),
			)),
			CurrencyId::Pendulum => Some(MultiLocation::new(
				1,
				X2(Parachain(PENDULUM_ID).into()), PalletInstance(10)),
			)),
			CurrencyId::Amplitude => Some(MultiLocation::new(
				1,
				X2(Parachain(AMPLITUDE_ID).into()), PalletInstance(10)),
			)),
			_ => None,
		}
	}
}
```